src/dlgTriggerEditor.cpp

```diff
@@ -4427,6 +4453,27 @@ void dlgTriggerEditor::selectKeyByID(int id)
     }
 }
 
+void dlgTriggerEditor::slot_triggerEdited()
+{
+    QTreeWidgetItem* item = mpCurrentTriggerItem;
+    if (!item) {
+        return;
+    }
+    if (!item->parent()) {
+        return;
+    }
+
+    const int triggerID = item->data(0, Qt::UserRole).toInt();
+    TTrigger* trigger = mpHost->getTriggerUnit()->getTrigger(triggerID);
+
+    const auto packageName = trigger->packageName(trigger);
+    if (packageName.isEmpty() || mpHost->mModuleInfo.contains(packageName)) {
+        return;
+    }
+
+    showWarning(tr("Note: any changes you make to the package will be lost during the upgrade. Consider making a copy of the trigger outside the package first."));
```

Comment:
Might I suggest this modification - which means the End-user gets a similar warning should they try and modify a module in a profile that is NOT set for it to sync (to/from other copies in other profiles) as then if it gets modified (or even just saved) in another profile then the local changes will get blown away:
```suggestion
    if (packageName.isEmpty()) {
        // This is not part of a package OR a module
        return;
    }

    const QStringList moduleDetails = mpHost->mInstalledModules.value(packageName);
    if (!moduleDetails.count()) {
        // This is NOT part of a module (there are no module details) - so it
        // is a package:
        showWarning(tr("Note: any changes you make to the package will be overwritten by an upgrade. Consider making a copy of the trigger outside the package first."));
    } else if (moduleDetails.count() > 1 && (moduleDetails.at(1).toInt() == 0)) {
        // It has the two entries required for a module and the second one
        // indicates that it is NOT synced (is a zero):
        showWarning(tr("Note: any changes you make to the module will be overwritten by a sync action from the same module in another profile. Consider making a copy of the trigger outside the module first."));
    }
    // Else it is a synced module - so changes will be saved
```

Reply:
There's no guarantee the same module will be installed in other profiles though? The code doesn't check for this. Would you like to fix that?

Reply:
:thinking: I'm not sure I follow you - if the module in this profile is set to be synced then any changes made to it will be saved (and propagated to other instances in other profiles whether loaded or not) however if it is NOT synced than any changes made will NOT be saved (between sessions) and should another active profile   with the same module that IS synced get saved than the local changes will (or should) be immediately lost/overwritten.

{This does seem to happen most of the time but whilst testing I got some weird cases where the changes in a "synced" profile was not saved to the file copy on disk - so didn't propagate to the other profiles :worried: - I am investigating further :toolbox::hammer:}.

Basically - what do you mean by "There's no guarantee the same module will be installed in other profiles though?" - there is no simple way to check if any other profile whether loaded or not uses the same module - but that is not something that this warning has to concern itself with as far as I can tell. If the module in this profile is NOT "synced" then changes made to it will not be preserved as the original, unmodified version will get loaded in from the file the next time the profile is loaded {exactly as is the case for if this was a package} anyhow.

Reply:
I think we are getting a lost in the weeds here. This niche edge case is challenging to think about, and more importantly, it is not an issue affecting folks. The issue affecting folks is in packages, not modules, and this PR addresses that.

Reply:
It can affect both - my guess is that those using modules haven't moaned loudly enough about this because they think it just happens like this (the same that you can put stuff into the brown top level folder for a module but it isn't saved because it is a "dummy"/virtual container - or that the "Delete" button is enabled for the top-level package or module folder but it does not/can not remove the item so should not be.).

I can confirm though that unsynced module have the same issue as this PR is trying to fix for packages so I think it is reasonable to address both in the same way.

Reply:
If you can find someone else on the team who understands the nuances of this reckons it's good to have, we can add it.

Reply:
It doesn't look like this gathered support in a couple of months, so lets try this version first